support animated image transformations

wjordan opened the pull request:
Support resizing animated image transformations using the existing pegasus dynamic-image routes.

example on local server:
http://localhost:3000/images/fit-x200/AR2017/pokemon.gif

Before:

![image](https://user-images.githubusercontent.com/56541/44613510-37216280-a7c9-11e8-81f0-5395102ee7bf.png)

After:
![image](http://cdo-dist.s3.amazonaws.com/pokemon%20(1).gif)

breville commented on lib/cdo/pegasus/graphics.rb:
```diff
@@ -119,3 +123,23 @@ def optimize_image(blob)
+  require 'image_optim'
+  gifsicle = ImageOptim.new.resolve_bin!(:gifsicle).path
+  resize_str = case mode
+               when :fill
```
Our linter likes this? :)

wjordan commented:
I wasn't too thrilled with manually shelling-out to `gifsicle` so I took another look at this. It looks like RMagick supports animated images (via [`ImageList`](https://rmagick.github.io/ilist.html)) directly, so I updated the PR to use this approach instead. I think the result is a bit simpler (rather than creating a second parallel image-processing routine, it uses the same one).

wjordan commented:
> What's the best way to verify this change?

I've added a commit with test coverage around these use cases.

>  Should I load up a large GIF through my development server and observe that it's smaller?

Not exactly- this PR only specifically fixes a limitation in the image-transformation Pegasus route, any other optimized-image routes already worked as expected for animated images.. Specifically, in the tests I've added the `"test_optimize_anigif"` test passes even before this PR, while the added `"assert_animated_image"` entry in `"test_process_image"` is fixed by this PR.

breville commented:
Looks like these can sometimes take too long?

e.g. https://code.org/images/fit-214x150/animated-examples/student-dance-projects/ryan_project.gif seems to timeout with a CloudFront 504 error.
